src/test/java/org/apache/commons/crypto/stream/CtrCryptoStreamTest.java

```diff
@@ -52,4 +75,115 @@ protected CtrCryptoOutputStream getCryptoOutputStream(
         }
         return new CtrCryptoOutputStream(baos, cipher, bufferSize, key, iv);
     }
+    
+    @Override
+    protected CtrCryptoOutputStream getCryptoOutputStream(final String transformation,
+            final Properties props, final ByteArrayOutputStream baos, final byte[] key, 
+            final AlgorithmParameterSpec params, final boolean withChannel) throws IOException {
+        if (withChannel) {
+            return new CtrCryptoOutputStream(props, Channels.newChannel(baos), key, 
+                    ((IvParameterSpec)params).getIV());
+        }
+        return new CtrCryptoOutputStream(props, baos, key,((IvParameterSpec)params).getIV());
+    }
+    
+    @Override
+    protected void doFieldGetterTest(final String cipherClass, final ByteArrayOutputStream baos,
+            final boolean withChannel) throws Exception {
+        if (AbstractCipherTest.OPENSSL_CIPHER_CLASSNAME.equals(cipherClass)) {
+            if (!Crypto.isNativeCodeLoaded()) {
+                return; // Skip this test if no JNI
+            }
+        }
+        
+        StreamInput streamInput = new StreamInput(new ByteArrayInputStream(encData), 0);
+        try {
+            streamInput.seek(0);
+            Assert.fail("Expected UnsupportedOperationException.");
+        } catch (UnsupportedOperationException ex) {
+        	Assert.assertEquals(ex.getMessage(), "Seek is not supported by this implementation");
```

Comment:
Isn't asserting the exception is generated enough?  The same goes for the below tests as well, why compare the error message?

Reply:
In this case I believe it would be enough.  IIRC, I was following an existing pattern for other tests which covered cases where different exceptions were caught, wrapped into a common exception and thrown as that more general exception, which was then only differentiated by the message.  I don't think it's hurting anything but I have no issue with you removing it.

Reply:
Actually, now that I think about it I'd prefer it remained in.  I think it helps with readability.

Reply:
I agree it helps with readability.  My reservation is this is a brittle test if the generation method changes in the future, all these tests have to be changed.

Reply:
I agree, but I think that's the point of the tests.  I would want a future developer to be conscious of the impact that changes had on the test and make necessary modifications, but I don't want to get into a philosophical discussion about software testing...

Reply:
Lets keep it as is then.

Reply:
LGTM